### ISSUE TITLE
Make configurable the mode where key mapping switches tabs

### ISSUE THREAD

**Issue by thyttan** [FabianWirth/search.nvim#19]
First, thanks for the plugin - I like it!

I'd prefer to switch between tabs like if they were buffers, i.e. with "H"/"L" respectively, having this mapping only act when in normal mode. 

With the current state I'm halfway there by changing the key mappings, but this stops me from being able to search for strings containing "H" or "L". This could be handled if I could enable the mapping only for normal mode.

Thanks again!

**Comment by Ev-1** [comment]
Hello, @thyttan. I'm not the maintainer, but I've made a PR with what I think you want, #20 . If you could test it to see if it has the behavior you want, that would be cool.

**Comment by thyttan** [comment]
Just switched to your fork/branch, and it works as I expect. Thanks!

I got this error message when quickly and repeatedly switching tabs:

```
   Error  22:26:18 msg_show.lua_error Error executing luv callback:
...local/share/nvim/lazy/telescope.nvim/lua/telescope/_.lua:137: 0
stack traceback:
	[C]: in function 'assert'
	...local/share/nvim/lazy/telescope.nvim/lua/telescope/_.lua:137: in function <...local/share/nvim/lazy/telescope.nvim/lua/telescope/_.lua:136>
	[C]: in function 'nvim_set_current_win'
	.../share/nvim/lazy/plenary.nvim/lua/plenary/popup/init.lua:408: in function 'create'
	...share/nvim/lazy/telescope.nvim/lua/telescope/pickers.lua:513: in function '_create_window'
	...share/nvim/lazy/telescope.nvim/lua/telescope/pickers.lua:111: in function 'mount'
	...share/nvim/lazy/telescope.nvim/lua/telescope/pickers.lua:535: in function 'find'
	...im/lazy/telescope.nvim/lua/telescope/builtin/__files.lua:392: in function 'v'
	...im/lazy/telescope.nvim/lua/telescope/builtin/__files.lua:621: in function 'v'
	.../nvim/lazy/telescope.nvim/lua/telescope/builtin/init.lua:578: in function <.../nvim/lazy/telescope.nvim/lua/telescope/builtin/init.lua:537>
	[C]: in function 'pcall'
	...e/.local/share/nvim/lazy/search.nvim/lua/search/init.lua:59: in function 'open_telescope'
	...e/.local/share/nvim/lazy/search.nvim/lua/search/init.lua:159: in function 'previous_tab'
	[string ":lua"]:1: in main chunk
```

But I have not looked yet if this also happens on FabianWirth/search.nvim.

**Comment by Ev-1** [comment]
I tried holding down the key switching keys and didn't manage to trigger the error. I don't think it is related to my changes, but could be a result of the fast switching.

From the trackeback it seems something went wrong in telescope when calling the "tele_func". 

Did it print the error then continue working as expected after the error was displayed?

**Comment by thyttan** [comment]
>Did it print the error then continue working as expected after the error was displayed?

Yes, at least if I didn't continue rapidly switching for a very long time (maybe 10-20 seconds). At which point, telescope and/or neovim would start to glitch.

**Comment by Ev-1** [comment]
Huh, strange. Does it happen independent of which/how many pickers you have in the collection? Does it fail on the same picker every time?

It seems I managed to trigger something odd now. The gui of search.nvim remained on screen, but the telescope UI disappeared, like this.
![image](https://github.com/FabianWirth/search.nvim/assets/33662061/010b5154-6cd8-426e-90e4-f53fb304de63)

Checking `:messages`

```lua
E5108: Error executing lua ...n/.local/share/nvim/lazy/search.nvim/lua/search/init.lua:54: BufLeave Autocommands for "<buffer=709>": Vim(append):Error executing lua callback: ...share/nvim/lazy/telescope.nvim/l
ua/telescope/pickers.lua:757: BufLeave Autocommands for "<buffer=709>"..WinLeave Autocommands for "<buffer=709>": Vim(append):Error executing lua callback: [NULL]
stack traceback:
        [C]: in function 'nvim_win_close'
        ...share/nvim/lazy/telescope.nvim/lua/telescope/pickers.lua:757: in function 'close_windows'
        ...share/nvim/lazy/telescope.nvim/lua/telescope/pickers.lua:1500: in function 'on_close_prompt'
        ...share/nvim/lazy/telescope.nvim/lua/telescope/pickers.lua:531: in function <...share/nvim/lazy/telescope.nvim/lua/telescope/pickers.lua:530>
        [C]: in function 'nvim_set_current_win'
        ...n/.local/share/nvim/lazy/search.nvim/lua/search/init.lua:54: in function 'open_telescope'
        ...n/.local/share/nvim/lazy/search.nvim/lua/search/init.lua:142: in function 'next_tab'
        [string ":lua"]:1: in main chunk
stack traceback:
        [C]: in function 'nvim_set_current_win'
        ...n/.local/share/nvim/lazy/search.nvim/lua/search/init.lua:54: in function 'open_telescope'
        ...n/.local/share/nvim/lazy/search.nvim/lua/search/init.lua:142: in function 'next_tab'
        [string ":lua"]:1: in main chunk
```

Managed to get another trace which failed at the same assert in telescope that you got earlier.

I think it might just be telescope not really being made for switching that fast.

**Comment by thyttan** [comment]
Don't know if your still interested, but just in case:

>Does it happen independent of which/how many pickers you have in the collection? Does it fail on the same picker every time?

Here's my configuration:
```
local builtin = require('telescope.builtin')
require("search").setup({
  mappings = {
    next = { { "L", "n" }, { "<Tab>", "n" }, { "<Tab>", "i" } },
    prev = { { "H", "n" }, { "<S-Tab>", "n" }, { "<S-Tab>", "i" } }
  },
--  mappings = { -- optional: configure the mappings for switching tabs (will be set in normal and insert mode(!))
--    next = "L",
--    prev = "H"
--  },
  append_tabs = { -- append_tabs will add the provided tabs to the default ones
    {
      "Commits", -- or name = "Commits"
      builtin.git_commits, -- or tele_func = require('telescope.builtin').git_commits
      available = function() -- optional
        return vim.fn.isdirectory(".git") == 1
      end
    },
  },
})
```

I'm happy to leave this at "I think it might just be telescope not really being made for switching that fast." :+1:

**Comment by thyttan** [comment]
Fixed with #20
